### PR TITLE
Fixes for ACE/bin/ace-install

### PR DESCRIPTION
**Pull request by okellogg**
- On running `find` for copying TAO libs and for copying ACE libs, prune directories `debian` and `tests`.
- On copying ACE executables change `ACE_PROGS` to `apps/gperf/src/ace_gperf`.

### CODE REVIEW

**Comment by mitza-oci** [comment]
Instead of "pruning out" directories, it may be easier (+ more maintainable) to root the search at the top-level directories you'd want.

**Comment by okellogg** [comment]
@mitza-oci I agree in principle.
Further, I don't understand why the lib-copying `find` was made as it was.
Why not just use
```
    for f in `find . -type f \( -name "lib?*.so*" -o -name "lib?*.a" \) -print`; do
```
to grab the shared and static libs?
(Perhaps extended by `"*.dll"` for Windows?)

**Comment by jwillemsen** [comment]
Feel free to update the find @okellogg, this script is 20 years old, it worked, so nobody touched it unless really required

**Comment by okellogg** [comment]
Thanks @jwillemsen.
I have [changed](https://github.com/DOCGroup/ACE_TAO/pull/1097/commits/f175070387a55026ba3df4571716c3822898f412) it as shown in my [previous comment](https://github.com/DOCGroup/ACE_TAO/pull/1097#issuecomment-622925822).
Caveat: I have not tested this on Windows; please give a shout if it breaks something.

**Comment by okellogg** [comment]
I wrote:
> Caveat: I have not tested this on Windows; please give a shout if it breaks something.

I at least tried now on Cygwin 3.1.4 and after manually postinstalling [cpio](https://cygwin.com/cgi-bin2/package-cat.cgi?file=x86_64%2Fcpio%2Fcpio-2.11-3&grep=cpio) and adding commit 796c2c0, it works there.
On doing so, in [config.h](https://github.com/DOCGroup/ACE_TAO/blob/master/ACE/ace/config-cygwin32.h) I had to add
```
#define ACE_HAS_CPU_SET_T 1
```
because I got a redefinition of type `cpu_set_t`.
The cpuset related stuff was touched in Cygwin 3.1, perhaps that has to do with it.

File: ACE/bin/ace-install [hunk 1]
```diff
@@ -438,7 +438,7 @@ if [ $install -ne 0 ]; then
 
     # copy TAO libs
     echo "Copying libraries..."
-    for f in `find . -type f -name "lib?*" -not -name "*\.dsw" -not -name "*Test*" -print`; do
+    for f in `find . -type f \( -name "lib?*.so*" -o -name "lib?*.a" -o -name "lib?*.dll" \) -print`; do
```
Comment (mitza-oci) [inline ACE/bin/ace-install:441]: `.dylib` for macOS

**Comment by mitza-oci** [comment]
Also take a look at the CI style/warning checks for shell
https://www.codefactor.io/repository/github/docgroup/ace_tao/pull/1097

File: ACE/bin/ace-install (hunk 1 above)
Comment (okellogg) [reply ACE/bin/ace-install]: Integrated in commit 0ef9468

**Comment by jwillemsen** [comment]
Please also don't use force pushes because they also require a full review because we can't tell what changed

**Comment by okellogg** [comment]
@mitza-oci wrote:
> [...] Take a look at MPC's install support, and (if using gnuace), the "make install" target which
> builds on MPC's install support.

If I do
```
cd TAO
$ACE_ROOT/bin/mwc.pl TAO_ACE.mwc -type gnuace
```
with an up-to-date [$MPC_ROOT](https://github.com/DOCGroup/MPC), the generated `TAO/GNUmakefile` does not have an `install` target (?)

**Comment by mitza-oci** [comment]
> @mitza-oci wrote:
> 
> > [...] Take a look at MPC's install support, and (if using gnuace), the "make install" target which
> > builds on MPC's install support.
> 
> If I do
> 
> ```
> cd TAO
> $ACE_ROOT/bin/mwc.pl TAO_ACE.mwc -type gnuace
> ```
> 
> with an up-to-date [$MPC_ROOT](https://github.com/DOCGroup/MPC), the generated `TAO/GNUmakefile` does not have an `install` target (?)

It's not supposed to have one in the text of that makefile.  It gets included from a different file.
The file ACE-INSTALL.html has instructions on using `make install` (search for that phrase).

**Comment by okellogg** [comment]
> [...] It gets included from a different file.
> The file ACE-INSTALL.html has instructions on using `make install`

Thanks! Got it.

**Pull request merged**